stan/math/gpu/err/check_matrix_gpu.hpp

```diff
@@ -0,0 +1,184 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
+#define STAN_MATH_PRIM_MAT_ERR_CHECK_GPU_HPP
+#ifdef STAN_OPENCL
+#include <stan/math/gpu/matrix_gpu.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <stan/math/prim/mat/err/constraint_tolerance.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <sstream>
+#include <string>
+
+namespace stan {
+namespace math {
+/**
+ * Check if the <code>matrix_gpu</code> is square.
+ *
+ * @param function Function name (for error messages)
+ * @param name Variable name (for error messages)
+ * @param y <code>matrix_gpu</code> to test
+ *
+ * @throw <code>std::invalid_argument</code> if the <code>matrix_gpu</code>
+ *    is not square
+ */
+inline void check_square(const char* function, const char* name,
```

Comment:
I think it makes sense that each of the functions are inside their own header file to parallel the rest of the math library. So please split these out.

Reply:
fixed